ts/Extensions/MouseWheelZoom/MouseWheelZoom.ts

```diff
@@ -66,198 +70,257 @@ const optionsToObject = (
+            innerStart = outerStart;
         } else {
-            inner.x = outer.x + outer.width - inner.width;
+            innerStart = outerStart + outerWidth - innerWidth;
         }
     }
-    if (inner.width > outer.width) {
-        inner.width = outer.width;
+    if (innerWidth > outerWidth) {
+        innerWidth = outerWidth;
     }
-    if (inner.x < outer.x) {
-        inner.x = outer.x;
+    if (innerStart < outerStart) {
+        innerStart = outerStart;
     }
 
-    // y and height
-    if (inner.y + inner.height > outer.y + outer.height) {
-        if (inner.height > outer.height) {
-            inner.height = outer.height;
-            inner.y = outer.y;
-        } else {
-            inner.y = outer.y + outer.height - inner.height;
-        }
-    }
-    if (inner.height > outer.height) {
-        inner.height = outer.height;
-    }
-    if (inner.y < outer.y) {
-        inner.y = outer.y;
-    }
-
-    return inner;
+    return {
+        rangeStart: innerStart,
+        rangeWidth: innerWidth
+    };
 };
 
 let wheelTimer: number,
     startOnTick: boolean|undefined,
     endOnTick: boolean|undefined;
 
 /**
+ * Temporarly disable `axis.startOnTick` and `axis.endOnTick` to allow zooming
+ * for small values.
  * @private
- */
-const zoomBy = function (
+*/
+const waitForAutomaticExtremes = function (
+    axis: Axis
+) : void {
+    const axisOptions = axis.options;
+
+    // Options interfering with yAxis zoom by setExtremes() returning
+    // integers by default.
+    if (defined(wheelTimer)) {
+        clearTimeout(wheelTimer);
+    }
+
+    if (!defined(startOnTick)) {
+        startOnTick = axisOptions.startOnTick;
+        endOnTick = axisOptions.endOnTick;
+    }
+
+    // Temporarily disable start and end on tick, because they would
+    // prevent small increments of zooming.
+    if (startOnTick || endOnTick) {
+        axisOptions.startOnTick = false;
+        axisOptions.endOnTick = false;
+    }
+    wheelTimer = setTimeout((): void => {
+        if (defined(startOnTick) && defined(endOnTick)) {
+            // Repeat merge after the wheel zoom is finished, #19178
+            axisOptions.startOnTick = startOnTick;
+            axisOptions.endOnTick = endOnTick;
+
+            // Set the extremes to the same as they already are, but now
+            // with the original startOnTick and endOnTick. We need
+            // `forceRedraw` otherwise it will detect that the values
+            // haven't changed. We do not use a simple yAxis.update()
+            // because it will destroy the ticks and prevent animation.
+            const { min, max } = axis.getExtremes();
+            axis.forceRedraw = true;
+            axis.setExtremes(min, max);
+            startOnTick = endOnTick = void 0;
+        }
+    }, 400);
+};
+
+/**
+* Calulate the ratio of mouse position on the axis to it's length. If mousePos
+* doesn't exist, returns 0.5;
+* @private
+*/
+const calucateFixToDirection = function (
```

Comment:
if this function calculates the ratio of mouse position, then I'd name the method something like `getMouseAxisRatio`, because the current name doesn't say anything.

Reply:
I agree, this doesn't say anything. I changed the method name and also the fixToDir variable name.